### DISCUSSION TITLE
Sort enums rule

### DISCUSSION THREAD

**Discussion by jpike88** [biomejs/biome#536] [Rule suggestion]
We have a few enums that are very large in size. Large enums without sort are a mess to look through.

The one has hundreds of keys.
<img width="825" alt="Screenshot 2023-10-17 at 16 07 22" src="https://github.com/biomejs/biome/assets/9585787/ba4c68b6-7e21-4407-acc9-17fd4daaa95a">

Would be good if a "sortEnumKeys" existed like "organiseImports", so that developers are forced to ensure the enum doesn't just more scrambled over time?

**Comment by steckdev** [comment]
I hope this message finds you well. I am an enthusiastic developer and an avid user of Biome. Recently, I came across the eslint-plugin-typescript-sort-keys plugin, which enforces sorting of interface and string enum keys in TypeScript. This functionality significantly improves code readability and maintainability, and I believe it would be a valuable addition to Biome's feature set.

Proposal
I would like to propose the integration of a TypeScript key sorting feature within Biome. This feature would include:

Interface Key Sorting: Ensuring that all keys in TypeScript interfaces are sorted alphabetically, with options to sort in ascending (asc) or descending (desc) order. Additional configurations could include case sensitivity, natural sorting, and prioritizing required keys.

String Enum Sorting: Similar functionality for string enums, ensuring that enum members are sorted alphabetically with customizable options.

Benefits
Enhanced Readability: Sorted keys make it easier for developers to navigate and understand code structures.
Consistency: Enforcing a consistent key order across a codebase reduces the likelihood of merge conflicts and simplifies code reviews.
Integration with Biome's Toolchain: Leveraging Biome's existing infrastructure for formatting and linting, this feature would provide a seamless and efficient developer experience.
Implementation
The implementation could involve:

Adding new linting rules for sorting TypeScript keys.
Providing configuration options within biome.json or via CLI commands.
Ensuring compatibility with existing formatting and linting features.
Contribution
I am keen to contribute to this feature and collaborate with the Biome community. Could we discuss the feasibility and any guidance or prerequisites for contributing to this functionality?

Thank you for considering this proposal. I look forward to your feedback and hope to assist in enhancing Biome's capabilities.

**Reply by Sec-ant** [reply]
I believe this kind of sorting rules belong to the `assists` scope, so please make sure to read #3178, #2652, #2412 if you want to contribute.

**Comment by nicholas-johnson-techxcel** [comment]
I cannot use this project without this. I will have to go back to eslint.

**Comment by Netail** [comment]
Related: https://perfectionist.dev/rules/sort-enums.html, we will probably pick this up with https://github.com/biomejs/biome/issues/9137 (as it can also be a GraphQL rule)

**Reply by Netail** [reply]
Soon: https://github.com/biomejs/biome/pull/10000 :)
